apps/system/js/bluetooth_transfer.js

```diff
@@ -643,60 +643,54 @@ var BluetoothTransfer = {
     var storageType = 'sdcard';
     var storage = navigator.getDeviceStorage(storageType);
     var getreq = storage.get(filePath);
+    var loader = LazyLoader.load(['shared/js/mime_mapper.js']);
+    Promise.all([getreq.then(), loader]).then(() => {
+      var file = getreq.result;
+      // When we got the file by storage type of "sdcard"
+      // use the file.type to replace the empty fileType which is given by API
+      var fileName = file.name;
+      var extension = fileName.split('.').pop();
+      var originalType = file.type || contentType;
+      var mappedType = (MimeMapper.isSupportedType(originalType)) ?
+        originalType : MimeMapper.guessTypeFromExtension(extension);
+
+      var a = new MozActivity({
```

Comment:
This also returns a DOMRequest :) you could chain this into the main chain if you want to.